### PR TITLE
Improve registration form.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Comment by mattcollier** [comment]
Added more protractor test to include testing the confirmPassphrase functions.

File: tests/protractor/pages/join.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+var element = GLOBAL.element;
+var should = GLOBAL.should;
+var expect = GLOBAL.expect;
+
+api.getJoin = function() {
+  bedrock.get('/join');
+  return api;
+};
+
+api.navigateToHomePage = function() {
```
Comment (dlongley) [inline tests/protractor/pages/join.js]: Seems unusual for this to be in the `join` page API. It should be on a `home` page API and this page (or others) could access it as needed.

File: tests/protractor/pages/join.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+var api = {};
+module.exports = api;
+
+var by = GLOBAL.by;
+var element = GLOBAL.element;
+var should = GLOBAL.should;
+var expect = GLOBAL.expect;
+
+api.getJoin = function() {
```
Comment (dlongley) [inline tests/protractor/pages/join.js]: Name this just `get` if it's scoped to the join page.

File: tests/protractor/pages/join.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  bedrock.get('/join');
+  return api;
+};
+
+api.navigateToHomePage = function() {
+  bedrock.get('/');
+  bedrock.waitForAngular();
+};
+
+api.login = function(options) {
```
Comment (dlongley) [inline tests/protractor/pages/join.js]: This should be on the API of another page, like a `navbar` page.

**Comment by mattcollier** [comment]
@dlongley revised based on your comments.

File: tests/protractor/pages/home.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * Identity provider home page.
+ *
+ * Copyright (c) 2015 The Open Payments Foundation. All rights reserved.
+ */
+var bedrock = GLOBAL.bedrock;
+var browser = GLOBAL.browser;
+var by = GLOBAL.by;
+var element = GLOBAL.element;
+var expect = GLOBAL.expect;
+var should = GLOBAL.should;
```
Comment (dlongley) [inline tests/protractor/pages/home.js]: Remove any vars that aren't used.

**Comment by mattcollier** [comment]
@dlongley revised based on your comments.

**Pull request merged**